application-admintools-default/src/main/java/com/xwiki/admintools/internal/wikiUsage/InstanceUsage.java

```diff
@@ -96,6 +114,35 @@ public String renderTemplate()
         }
     }
 
+    /**
+     * Retrieves in descending order the documents that have more than a given number of comments.
+     *
+     * @param maxComment maximum number of comments below which the document is ignored.
+     * @return a {@link List} with the documents that have more than the given number of comments.
+     * @throws QueryException if the query to retrieve the document fails.
+     * @throws XWikiException if a document is not found.
+     */
+    public List<XWikiDocument> getDocumentsOverGivenNumberOfComments(int maxComment)
+        throws QueryException, XWikiException
+    {
+        List<String> documentsWithComments = this.queryManager.createQuery(
+            "select obj.name from BaseObject obj where obj.className='XWiki.XWikiComments' group by obj.name",
+            Query.XWQL).setWiki(wikiDescriptorManager.getCurrentWikiId()).execute();
+
+        List<XWikiDocument> documentsOverMaxComments = new ArrayList<>();
+        for (String documentName : documentsWithComments) {
+            XWikiContext wikiContext = wikiContextProvider.get();
+            XWiki wiki = wikiContext.getWiki();
+            XWikiDocument document = wiki.getDocument(resolver.resolve(documentName), wikiContext);
+            int numberOfComments = document.getComments().size();
+            if (numberOfComments > maxComment) {
+                documentsOverMaxComments.add(document);
+            }
+        }
+        documentsOverMaxComments.sort((d1, d2) -> Integer.compare(d2.getComments().size(), d1.getComments().size()));
```

Comment:
I would do this from the query. For example, I used this HQL query for pages with more than 2 comments:
`select obj.name as name, count(*) as no from BaseObject obj where obj.className='XWiki.XWikiComments' group by obj.name having count(*) > 2 order by no desc`
It might need some adjustments, as well as having the `maxComments` value binded